test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,597 @@
+    if not (isinstance(input, datapoints.Image) and input.dtype is torch.uint8):
+        return
+
+    kernel = getattr(F, f"{dispatcher.__name__}_image_pil")
+
+    with mock.patch(f"{dispatcher.__module__}.{kernel.__name__}", wraps=kernel) as spy:
+        output = dispatcher(F.to_image_pil(input), *args, **kwargs)
+
+        spy.assert_called_once()
+
+    assert isinstance(output, PIL.Image.Image)
+
+
+def check_dispatcher(
+    dispatcher,
+    kernel,
+    input,
+    *args,
+    check_scripted_smoke=True,
+    check_dispatch_simple_tensor=True,
+    check_dispatch_datapoint=True,
+    check_dispatch_pil=True,
+    **kwargs,
+):
+    with mock.patch("torch._C._log_api_usage_once", wraps=torch._C._log_api_usage_once) as spy:
+        dispatcher(input, *args, **kwargs)
+
+        spy.assert_any_call(f"{dispatcher.__module__}.{dispatcher.__name__}")
+
+    unknown_input = object()
+    with pytest.raises(TypeError, match=re.escape(str(type(unknown_input)))):
+        dispatcher(unknown_input, *args, **kwargs)
+
+    if check_scripted_smoke:
+        _check_dispatcher_scripted_smoke(dispatcher, input, *args, **kwargs)
+
+    if check_dispatch_simple_tensor:
+        _check_dispatcher_dispatch_simple_tensor(dispatcher, kernel, input, *args, **kwargs)
+
+    if check_dispatch_datapoint:
+        _check_dispatcher_dispatch_datapoint(dispatcher, kernel, input, *args, **kwargs)
+
+    if check_dispatch_pil:
+        _check_dispatcher_dispatch_pil(dispatcher, input, *args, **kwargs)
+
+
+def _check_dispatcher_kernel_signature_match(dispatcher, *, kernel, input_type):
+    dispatcher_signature = inspect.signature(dispatcher)
+    dispatcher_params = list(dispatcher_signature.parameters.values())[1:]
+
+    kernel_signature = inspect.signature(kernel)
+    kernel_params = list(kernel_signature.parameters.values())[1:]
+
+    # We filter out metadata that is implicitly passed to the dispatcher through the input datapoint, but has to be
+    # explicit passed to the kernel.
+    kernel_params = [param for param in kernel_params if param.name not in input_type.__annotations__.keys()]
+
+    dispatcher_params = iter(dispatcher_params)
+    for dispatcher_param, kernel_param in zip(dispatcher_params, kernel_params):
+        try:
+            # In general, the dispatcher parameters are a superset of the kernel parameters. Thus, we filter out
+            # dispatcher parameters that have no kernel equivalent while keeping the order intact.
+            while dispatcher_param.name != kernel_param.name:
+                dispatcher_param = next(dispatcher_params)
+        except StopIteration:
+            raise AssertionError(
+                f"Parameter `{kernel_param.name}` of kernel `{kernel.__name__}` "
+                f"has no corresponding parameter on the dispatcher `{dispatcher.__name__}`."
+            ) from None
+
+        assert dispatcher_param == kernel_param
+
+
+def _check_dispatcher_datapoint_signature_match(dispatcher):
```

Comment:
> This check is called for every single parametrization of `test_dispatcher_signature`

Yes

> but with the same input, right?

Yes, but. Maybe I'm being pedantic here, but we don't actually have an "input" here, but rather only an `input_type`. Meaning, we never materialize the tensor.

Reply:
the input is the `dispatcher`, which is constant across all the parametrizations

Reply:
Ok, got your point now. You are right, that is somewhat wasteful. But I think this check is superfast since we don't materialize anything and so not sure if we want to act on this now.